src/dynamicprompts/parser/parse.py

```diff
@@ -378,11 +381,15 @@ def create_parser(
         parser_config=parser_config,
         prompt=variant_prompt,
     )
+    
+    variable_ref = pp.Combine(_configure_variable_access(parser_config=parser_config, prompt=pp.SkipTo(parser_config.variable_end))).leave_whitespace()
+    variable_ref.set_parse_action(lambda s,l,t: parser_config.variable_start + "".join(t) + parser_config.variable_end)
```

Comment:
Could we get some comments here? I can't immediately grok what the parse action does :D

Reply:
Added a comment. Essentially this just puts the variable back into the "${}" wrapper so it will be recognized as a variable again during sampling.